packages/zowe-explorer-api/src/vscode/ui/utils/TableMediator.ts

```diff
@@ -0,0 +1,102 @@
+ * This class serves to satisfy three main use cases:
+ *
+ * @remarks
+ *
+ * ## Adding a table
+ *
+ * Tables are added to the mediator based on their unique ID using {@link TableMediator.addTable}. The ID is in the following format:
+ *
+ * `<TableName>-<8digit-Unique-Id>##<ExtensionId>`
+ *
+ * This avoids indirect naming conflicts by making the table identifier specific to the contributing extension.
+ * The table ID can be accessed directly from a {@link Table.View} instance using the {@link Table.View.getId} function.
+ *
+ * ## Accessing a table
+ *
+ * Tables are only accessible by their unique IDs using {@link TableMediator.getTable}. Extenders must communicate a table ID
+ * to another extender to facilitate table changes between extensions. This facilitates explicit access requests
+ * by ensuring that two or more extenders are coordinating changes to the same table.
+ *
+ * The map containing the tables is not publicly exposed and is defined as a
+ * [private class property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties)
+ * to avoid exposing any tables to extenders who do not have explicit access.
+ *
+ * ## Removing a table
+ *
+ * Tables can only be removed by the extender that has contributed them using {@link TableMediator.deleteTable}.
+ * This establishes a read-only relationship between the mediator and extenders that have not contributed the table they are trying to access.
+ *
+ * **Note** that this does not prevent an extender with access to the ID from disposing the table, once they've received access to the instance.
+ */
+
+export class TableMediator {
+    private static instance: TableMediator;
+    private tables: Map<string, Table.View> = new Map();
+
+    private constructor() {}
+
+    /**
+     * Access the singleton instance.
+     *
+     * @returns the global {@link TableMediator} instance
+     */
+    public static getInstance(): TableMediator {
+        if (!this.instance) {
+            this.instance = new TableMediator();
+        }
+
+        return this.instance;
+    }
+
+    /**
+     * Adds a table to the mediator to enable sharing between extensions.
+     *
+     * @param table The {@link Table.View} instance to add to the mediator
+     */
+    public addTable(table: Table.View): void {
+        this.tables.set(table.getId(), table);
+    }
+
+    /**
+     * Accesses a table in the mediator based on its unique ID.
+     *
+     * @param id The unique identifier for the desired table
+     *
+     * @returns
+     * * {@link Table.View} instance if the table exists
+     * * `undefined` if the instance was deleted or does not exist
+     */
+    public getTable(id: string): Table.View | undefined {
+        return this.tables.get(id);
+    }
+
+    /**
+     * Removes a table from the mediator.
+     *
+     * @param id The unique ID of the table to delete
+     * @returns `true` if the table was deleted; `false` otherwise
```

Comment:
No changes requested here

---

I'm always conflicted about the return value of a delete-like method.
Depending on the use case:
- Someone might appreciate getting the deleted value back (like a `Stack.pop` operation)
- Or someone else might want `true` if the resource (table) didn't exist.

That said, I don't believe we use the return value of removeTable in the TableView.
So, it's fine to keep as is 😋

Reply:
I could see value in both return types - this was modeled after the `Map.prototype.delete()` function, which returns `true/false` when deleting a key from the map. I figured that most users will not want the table view after removing it - in the case where they want to quickly create table views with a similar structure, they can hold onto a builder and then re-define its data before calling `TableBuilder.build`.